edward/inferences/replica_exchange_mc.py

```diff
@@ -0,0 +1,284 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import six
+import tensorflow as tf
+import numpy as np
+
+from collections import OrderedDict
+from edward.inferences.monte_carlo import MonteCarlo
+from edward.models import Empirical, RandomVariable
+from edward.util import check_latent_vars, copy
+
+
+class memory_lambda:
+  """
+  Class to use instead of lambda.
+  lambda is affected by the change of x,
+  so memory_lambdda output x at the time of definition.
+  """
+
+  def __init__(self, x):
+    self.x = x
+
+  def __call__(self):
+    return self.x
+
+
+class ReplicaExchangeMC(MonteCarlo):
+  """
+  Replica Exchange MCMC(https://en.wikipedia.org/wiki/Parallel_tempering)
```

Comment:
Cool! Is there a canonical citation we can use instead of the Wiki article?

Reply:
I think we should cite Swendsen and Wang(1986), Hukushima and Nemoto(1996).
I'll add the citations to bib.bib and this code.